pages/curriculum.tsx

```diff
@@ -13,8 +15,32 @@ export const Curriculum: React.FC<GetAppProps> = ({ data }) => {
   const { loading, error, alerts, lessons, session } = data
 
   if (loading) return <LoadingSpinner />
-  if (error) return <h1>Error</h1>
-  if (!session || !lessons || !alerts) return <h1>Bad Data</h1>
+  if (
+    error &&
+    _.get(error, 'graphQLErrors[0].message') ===
+      "Cannot read property 'id' of null"
+  ) {
+    Router.push('/login')
+    return <LoadingSpinner />
+  }
+  if (error) {
+    return (
+      <Error
+        title="Internal server error"
+        message={error.message}
+        src="/500.png"
+      ></Error>
+    )
+  }
+  if (!session || !lessons || !alerts) {
+    return (
+      <Error
+        title="Internal server error"
+        message="Bad data"
+        src="/500.png"
+      ></Error>
+    )
+  }
```

Comment:
Instead of two `if statements`, you can use just one in order to reduce duplicated code
```
if (error || !session || !lessons || !alerts) {
    return (
      <Error
        title="Internal server error"
        message={error && error.message || "Bad data"}
        src="/500.png"
      ></Error>
    )
  }
```